### ISSUE TITLE
#compactSymbolTable leads to hash collisions

### ISSUE THREAD

**Issue by MarcusDenker** [guillep/pharo-core#11319]
Some simplistic benchmarks... 

Just after image startup of the downloaded image:

```
['asdasdf' asSymbol ] bench. "'1178179.564 per second'" 
```

Now do "Symbol compactSymbolTable":

```
['asdasdf' asSymbol ] bench. "'652087.348 per second'"
```

But if instead, we do a rehash: "Symbol rehash", which adds all instances to a new set (which leads
to growing naturally to a much larger size than 25%):

```
['asdasdf' asSymbol ] bench.   "'2083532.094 per second'"
```

**Comment by svenvc** [comment]
Good observation.

But this benchmark creates the symbol once and then finds it millions of times as already existing, so you are basically testing hashtable/dictionary lookup, which is apparently better when there is some free space.

Still, it is important.

**Comment by svenvc** [comment]
I would indeed try this

**Comment by MarcusDenker** [comment]
I merged #11320, that is, #compactSymbolTable is not used anymore

(we later should deprecate it, with calling #rehash instead)
